pkg/operations/handlers.go

```diff
@@ -119,7 +119,7 @@ func postProcessBlockGroups(blockGroups []*blockGroup) *blockListResult {
 	maxBlocksPerGroup := 0
 	for i := 0; i < len(blockGroups); i += 1 {
 		blockGroup := blockGroups[i]
-		if i < len(blockGroups)-1 && !strings.Contains(blockGroup.FormattedMinTime, "0:00Z") {
+		if i < len(blockGroups)-1 && shouldAddToNextGroup(blockGroup.FormattedMinTime) {
```

Comment:
Wouldn't it be better if we group them into equal hours (or half hours), rather than messing with strings?

```
minTime.Truncate(time.Hours)
```

Reply:
Thanks @simonswine. Switched to using `pyroscopedb.max-block-duration` to avoid having to "guess" the block duration, then truncating and grouping on this instead of strings. Compacted blocks can have a different duration but this still works well enough for now.

Reply:
Yes that's even better, all compactor block ranges will have to be multiples of the max-block-duration.